sklearn_numba_dpex/utils/tests/test_device_params.py

```diff
@@ -0,0 +1,12 @@
+import dpctl
+
+
+def test_opencl_requirement():
+    device = dpctl.select_default_device()
+    has_preferred_work_group_size_multiple = hasattr(
+        device, "preferred_work_group_size_multiple"
+    )
+    has_global_mem_cache_size = hasattr(device, "global_mem_cache_size")
+    assert not (
+        has_preferred_work_group_size_multiple or has_global_mem_cache_size
+    ), "pyopencl is not required anymore, use dpctl instead"
```

Comment:
I don't understand this assertion message.

Reply:
Ah ok this test is a detector for the TODO in https://github.com/soda-inria/sklearn-numba-dpex/pull/2/files#diff-cd7bbe3194128114b8a2f3fe52d4f4db8d1d8a702ad95c1971e4a863c595a183R4.

Instead we should update the code in `sklearn_numba_dpex/utils/_device.py` to directly use `preferred_work_group_size_multiple` when available.

Reply:
> Instead we should update the code in `sklearn_numba_dpex/utils/_device.py` to directly use `preferred_work_group_size_multiple` when available.

it's like this already, but I thought this test could be a reminder that we can remove the dependency to pyopencl (or start a deprecation cycle)

Reply:
Also, could you please write a test for `_get_cl_param` and check that it raises a warning only when the device cannot be found via pyopencl?

Reply:
done